lld/test/MachO/icf-safe-thunks.ll

```diff
@@ -49,127 +62,179 @@
 ; CHECK-ARM64-MAP-NEXT: 0x00000004 [  2] _func_3identical_v2
 ; CHECK-ARM64-MAP-NEXT: 0x00000004 [  2] _func_3identical_v3
 
+; ModuleID = 'icf-safe-thunks.cpp'
+source_filename = "icf-safe-thunks.cpp"
 target datalayout = "e-m:o-i64:64-i128:128-n32:64-S128-Fn32"
 target triple = "arm64-apple-macosx11.0.0"
-
-@g_val = global i8 0, align 1
-@g_ptr = global ptr null, align 8
-
+@g_val = global i8 0, align 1, !dbg !0
+@g_ptr = global ptr null, align 8, !dbg !7
 ; Function Attrs: mustprogress nofree noinline norecurse nounwind ssp memory(readwrite, argmem: none) uwtable(sync)
-define void @func_unique_1() #0 {
+define void @func_unique_1() #0 !dbg !19 {
```

Comment:
It seems the new test included many debug locations. Are they really necessary? If not, can we remove them to show with better clarity how the test changed with this patch?

Reply:
Tried for a while but it's very tricky manually editing out the debug info and still getting a valid dSYM that gives us the info we want. We don't typically edit out debug info as it's tricky to get it right. 

Does the (commented) source code at the bottom help understanding the test ? I think that would be a better guide, the only reason we include .ll here is because we can't invoke clang - otherwise the test would just be the C++ code.

On the C++ part the only thing that changed about the code is the addition of `-g` to the clang compile side. If needing to further investigate stripping debug info - is there any in particular that you had in mind ?

Reply:
You can use `llvm-reduce` to reduce your test case down to its simplest reproducer and removes unnecessary lines. This makes the test less distracting and easier to understand.

https://llvm.org/docs/CommandGuide/llvm-reduce.html

Reply:
I've also seen `update_llc_test_checks.py` used, but I haven't played with it yet. I think it will automatically generate `CHECK` lines for you.

https://llvm.org/docs/TestingGuide.html#generating-assertions-in-regression-tests

Reply:
I used a fair amount of `update_llc_test_checks.py`, it works pretty nice. But I'm not sure if it works with lld?

Reply:
I am not sure llvm-reduce would help here. The test is an existing minimal test that checks functionality - it can't be reduced any further without removing scenarios that need to be covered - i.e. it's preventing regressions rather than trying to reproduce a specific bug. 

This change just adds debug info to the test in order to repro the bug. 

I guess the alternative would be to create a new separate test that just reproduces the bug and doesn't also serve as a feature test. Would that be better ?

Reply:
I've reduced it to minimal repro.

Reply:
Great! Thanks